Lib/plistlib.py

```diff
@@ -217,14 +218,25 @@ def get_data(self):
 
     def begin_dict(self, attrs):
         d = self._dict_type()
+        self.previous_dict_key = self.current_key
         self.add_object(d)
         self.stack.append(d)
 
     def end_dict(self):
         if self.current_key:
             raise ValueError("missing value for key '%s' at line %d" %
                              (self.current_key,self.parser.CurrentLineNumber))
-        self.stack.pop()
+        d = self.stack.pop()
+        # Unmarshal CF$UID to UID and replace the dict node
+        if len(d) == 1 and "CF$UID" in d and isinstance(d["CF$UID"], int):
```

Comment:
Since `"CF$UID"` ends up returned by the public API of this module, it might be a good idea to extract this value to an importable constant